Research/lva-node-app/app.js

```diff
@@ -0,0 +1,312 @@
+        registry.getModulesOnDevice(DEVICE_ID).then((response) =>
+        {
+            let modules = "";
+            //get modules on device
+            for (let i = 0; i < response.responseBody.length; i++)
+            {
+                modules += response.responseBody[i].moduleId + " ";
+                console.log(response.responseBody[i].moduleId);
+            }
+            //on success
+            res.send("Successfully connected to your IoTHub. Found the following modules on your device: " + modules);
+        }).catch((reason) =>
+        {
+            console.error(reason);
+            res.status(400).send("Unable to connect to device. Error response: " + JSON.parse(reason.responseBody).Message);
+        });
+    }).catch((error) =>
+    {
+        console.error(error);
+        res.status(400).send("Error: " + error);
+    })
+})
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// defined functions
+
+/**
+* set device ID and IoTHub connection string
+* try to connect to the iot hub to make sure that the passed in values are valid!
+* @returns {Promise<any>} - Promise to resolve/reject based on validity of Device ID and Iot Hub Connection String
+*/
+function setConfigs(req)
+{
+    return new Promise((resolve, reject) =>
+    {
+        DEVICE_ID = req.body["device-id"];
+        IOTHUB_CONNECTION_STRING = req.body["iothub-connection-string"];
+        if (DEVICE_ID == "" || IOTHUB_CONNECTION_STRING == "")
+        {
+            DEVICE_ID = "";
+            IOTHUB_CONNECTION_STRING = "";
+            reject("You must provide both your device ID and your Iot Hub Connection String");
+        }
+        convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+        {
+            IOTHUB_ENDPOINT = eventhubString;
+            resolve();
+        }).catch((error) =>
+        {
+            //reset values on failure
+            console.error(error);
+            DEVICE_ID = "";
+            IOTHUB_CONNECTION_STRING = "";
+            reject(error.message);
+        });
+    })
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) =>
+{
+    wss.clients.forEach((client) =>
+    {
+        if (client.readyState === WebSocket.OPEN)
+        {
+            try
+            {
+                client.send(data);
+            } catch (error)
+            {
+                console.error(error);
+            }
+        }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+// stop old one start new or just check if it's already running
+*/
+function receiveHubMessages()
+{
+    const consumerGroup = "$Default"; // name of the default consumer group
+    if (eventHubReader == undefined)
```

Comment:
stop looks fixed, still an issue here i think unless startReadMessage is idempotent? if start is called twice it won't be undefined the 2nd time, but we will call start again. is that okay?

Reply:
Agh, duplicate messages. Modified appropriately, issue is taken care of in back end now!